envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadataValidator.kt

```diff
@@ -53,35 +53,39 @@ class NodeMetadataValidator(
 
     override fun onStreamOpen(streamId: Long, typeUrl: String?) {}
 
-    override fun onV3StreamRequest(streamId: Long, request: v3DiscoveryRequest?) {
+    override fun onV3StreamRequest(streamId: Long, request: DiscoveryRequestV3?) {
         request?.node?.let { validateV3Metadata(it) }
     }
 
-    override fun onV2StreamRequest(streamId: Long, request: DiscoveryRequest?) {
+    override fun onV2StreamRequest(streamId: Long, request: DiscoveryRequestV2?) {
         request?.node?.let { validateV2Metadata(it) }
     }
 
     override fun onStreamResponse(
         streamId: Long,
-        request: DiscoveryRequest?,
+        request: DiscoveryRequestV2?,
         response: DiscoveryResponse?
     ) {
     }
 
-    private fun validateV3Metadata(node: io.envoyproxy.envoy.config.core.v3.Node) {
+    private fun validateV3Metadata(node: NodeV3) {
         // Some validation logic is executed when NodeMetadata is created.
         // This may throw NodeMetadataValidationException
         val metadata = NodeMetadata(node.metadata, properties)
 
         validateMetadata(metadata)
     }
 
-    private fun validateV2Metadata(node: Node) {
+    private fun validateV2Metadata(node: NodeV2) {
         // Some validation logic is executed when NodeMetadata is created.
         // This may throw NodeMetadataValidationException
         val metadata = NodeMetadata(node.metadata, properties)
```

Comment:
I'd move it into the if.

Reply:
If we move it into the if then it `serviceName` won't be available to the exception.

Reply:
`NodeMetadataValidationException` message main purpose is to be shown on the client side (in envoy logs). We don't need a serviceName on that side. Is `NodeMetadataValidationException` also logged on the EC side? I don't remember.

Reply:
I think Exception would not be logged so information about isn't necessary.

Reply:
Removed